src/main/java/com/conveyal/datatools/manager/controllers/api/LabelController.java

```diff
@@ -0,0 +1,218 @@
+import static spark.Spark.*;
+import static spark.Spark.post;
+
+
+public class LabelController {
+    private static final Logger LOG = LoggerFactory.getLogger(LabelController.class);
+    private static JsonManager<Label> json = new JsonManager<>(Label.class, JsonViews.UserInterface.class);
+
+
+    /**
+     * Spark HTTP endpoint to get a single label by ID
+     */
+    private static Label getLabel(Request req, Response res) {
+        return requestLabelById(req, Actions.VIEW);
+    }
+
+    /**
+     * Spark HTTP endpoint that handles getting all labels for a handful of use cases:
+     * - for a single project (if projectId query param provided)
+     */
+    private static Collection<Label> getAllLabels(Request req, Response res) {
+        Auth0UserProfile user = req.attribute("user");
+        String projectId = req.queryParams("projectId");
+        Project project = Persistence.projects.getById(projectId);
+        if (project == null) {
+            logMessageAndHalt(req, 400, "Must provide valid projectId query param to retrieve labels.");
+        }
+
+        boolean isProjectAdmin = user.canAdministerProject(project.id, project.organizationId);
+
+        return project.retrieveProjectLabels().stream()
+                // Only return labels user has access to
+                .filter(label -> isProjectAdmin || !label.adminOnly)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * HTTP endpoint to create a new label
+     */
+    private static Label createLabel(Request req, Response res) throws IOException {
+        Auth0UserProfile userProfile = req.attribute("user");
+        Label newLabel = getPOJOFromRequestBody(req, Label.class);
+        validate(req, newLabel);
+        // User may not be allowed to create a new label
+        checkLabelPermissions(req, newLabel, Actions.CREATE);
+
+        try {
+            Persistence.labels.create(newLabel);
+            return newLabel;
+        } catch (Exception e) {
+            logMessageAndHalt(req, 500, "Unknown error encountered creating label", e);
+            return null;
+        }
+    }
+
+    /**
+     * Check that updated or new label object is valid. This method should be called before a label is
+     * persisted to the database.
+     */
+    private static void validate(Request req, Label label) {
+        List<String> validationIssues = new ArrayList<>();
+        // Label is quite forgiving (sets defaults if null) and the boolean value is type checked,
+        // so there is little to validate.
+        if (StringUtils.isEmpty(label.name)) {
+            validationIssues.add("Name field must not be empty.");
+        }
+        if (!validationIssues.isEmpty()) {
+            logMessageAndHalt(
+                    req,
+                    HttpStatus.BAD_REQUEST_400,
+                    "Request was invalid for the following reasons: " + String.join(", ", validationIssues)
+            );
+        }
```

Comment:
Because only a single label is being checked, I think this method can be simplified to something like:     

```
        // Label is quite forgiving (sets defaults if null) and the boolean value is type checked,
        // so there is little to validate.
        if (StringUtils.isEmpty(label.name)) {
            logMessageAndHalt(
                req,
                HttpStatus.BAD_REQUEST_400,
                "Request was invalid, the name field must not be empty."
            );
        }

```

Reply:
This is much cleaner thank you!